### ISSUE TITLE
A packager/release tool

### ISSUE THREAD

**Issue by maxtaco** [maxtaco/coffee-script#180]
The tool should be written in `iced`, meaning we assume that we currently have a global working version of `iced`, which isn't too bad an assumption.

There should be one additional source file which specifies the `iced` version on top of the current CoffeeScript version forked from master.  From this version, we can produce the `iced` version, which is:

```
<100*CoffeeScript.Major + CoffeeScript.Minor>. <CoffeeScript.Patch> . <IcedCoffeeScript.Patch>
```

The script should:
- Make a release branch with the release version
- Compile all of the sources
- Commit 
- Run tests
- Tag release
- Run `npm publish`

**Comment by zapu** [comment]
So the compiled release files will be in the same branch, but we will only commit them for the release? Kind of like it has been done so far. Do I understand this correctly?

**Comment by maxtaco** [comment]
Yeah, exactly.  I think we need a tool to automate all of this, otherwise, we're bound to make mistakes.

**Comment by zapu** [comment]
Hm, so I don't think I understand the forking completely. Couldn't it just work on current commit in current git repository?

**Comment by maxtaco** [comment]
Sorry, fork was the wrong word, sorry about that.

**Comment by zapu** [comment]
I'd still like to revisit the idea of keeping compiled files in separate branch, but maybe that's for later - I'm not sure how this could be done right now. Thanks for clarifying! I'll get it done.

**Comment by maxtaco** [comment]
The big issue with checking in the compiled code I've found is that it makes merging really hairy.  Not impossible to deal with, just hairy...

**Comment by zapu** [comment]
There should probably be a script to help with merges with mainline as well, what do you think?

**Comment by zapu** [comment]
So the additional source file would specify iced version number (single number, the `IcedCoffeeScript.patch`). But where would the CoffeeScript version be obtained from?

**Comment by maxtaco** [comment]
Can we read it directly out of `package.json`?  ~~If not, it's exported from `coffee-script.coffee`~~

**Comment by zapu** [comment]
Oh, right. Thanks
